### DISCUSSION TITLE
Redeclaring constraints using opti-stack takes inceasingly long for each MPC-iteration

### DISCUSSION THREAD

**Discussion by yngveki** [casadi/casadi#3209] [Q&A]
I have implemented an MPC-application in Python 3.10.6 where I use the opti-stack for declaring the cost function, the constraints and solving the optimization problem by calling IPOPT. 

Due to the nature of my MPC problem formulation, I need to update the constraints at every MPC-iteration to account for changes in historic values of actuation values and measured outputs from my system. In order to clear the previous iteration's constraints before I add the updated set of constraints, I call 

```
opti.subject_to()
opti.subject_to(constraints)
```

where `constraints` is the list of updated constraints.

After some debugging, I found that calling `opti.subject_to(constraints)` takes ~2-5ms longer for each iteration. Has anyone tackled this problem before? Am I mistaken in that I should redeclare the constraints at each iteration?

**Comment by jgillis** [comment]
Modifying the problem online is not ideal, although the cost should indeed not rise. Can you modify the problem to use opti.parameter instead, avoiding a need for online structural modification?

**Reply by yngveki** [reply]
I probably could, though I do not have the time before in a little while. It is not detrimental to my MPC that redeclaring the constraints costs increasing computational resources. Nevertheless, I will report back if I get time to fix it. Thanks for a speedy response!
